lib/auth/healthcheckconfig/healthcheckconfigv1/service.go

```diff
@@ -0,0 +1,192 @@
+		return nil, trace.Wrap(err)
+	}
+
+	created, err := s.backend.CreateHealthCheckConfig(ctx, req.GetConfig())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	s.emitAuditEvent(ctx, newCreateAuditEvent(ctx, created))
+	return created, nil
+}
+
+// GetHealthCheckConfig returns the specified HealthCheckConfig.
+func (s *Service) GetHealthCheckConfig(ctx context.Context, req *healthcheckconfigv1.GetHealthCheckConfigRequest) (*healthcheckconfigv1.HealthCheckConfig, error) {
+	authCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := authCtx.CheckAccessToKind(types.KindHealthCheckConfig, types.VerbRead); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	cfg, err := s.backend.GetHealthCheckConfig(ctx, req.GetName())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return cfg, nil
+}
+
+// ListHealthCheckConfigs lists HealthCheckConfig resources.
+func (s *Service) ListHealthCheckConfigs(ctx context.Context, req *healthcheckconfigv1.ListHealthCheckConfigsRequest) (*healthcheckconfigv1.ListHealthCheckConfigsResponse, error) {
+	authCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := authCtx.CheckAccessToKind(types.KindHealthCheckConfig, types.VerbRead, types.VerbList); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	page, token, err := s.backend.ListHealthCheckConfigs(ctx, int(req.GetPageSize()), req.GetPageToken())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &healthcheckconfigv1.ListHealthCheckConfigsResponse{
+		Configs:       page,
+		NextPageToken: token,
+	}, nil
+}
+
+// UpdateHealthCheckConfig updates an existing HealthCheckConfig.
+func (s *Service) UpdateHealthCheckConfig(ctx context.Context, req *healthcheckconfigv1.UpdateHealthCheckConfigRequest) (*healthcheckconfigv1.HealthCheckConfig, error) {
+	authCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := authCtx.CheckAccessToKind(types.KindHealthCheckConfig, types.VerbUpdate); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	updated, err := s.backend.UpdateHealthCheckConfig(ctx, req.GetConfig())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	s.emitAuditEvent(ctx, newUpdateAuditEvent(ctx, updated))
+	return updated, nil
+}
+
+// UpsertHealthCheckConfig creates or replaces a HealthCheckConfig.
+func (s *Service) UpsertHealthCheckConfig(ctx context.Context, req *healthcheckconfigv1.UpsertHealthCheckConfigRequest) (*healthcheckconfigv1.HealthCheckConfig, error) {
+	authCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := authCtx.CheckAccessToKind(types.KindHealthCheckConfig, types.VerbCreate, types.VerbUpdate); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	upserted, err := s.backend.UpsertHealthCheckConfig(ctx, req.GetConfig())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	s.emitAuditEvent(ctx, newCreateAuditEvent(ctx, upserted))
```

Comment:
nit: will tctl call Upsert or Update? would be nice to tell the difference here for the audit log.